L1Trigger/L1THGCal/src/be_algorithms/HGCalTriggerCellCalibration.cc

```diff
@@ -33,7 +33,13 @@ void HGCalTriggerCellCalibration::calibrateInMipT(l1t::HGCalTriggerCell& trgCell
     }
 
     /* correct the charge amplitude for the sensor thickness */
-    double trgCellMipP = amplitude / thickCorr_.at( cellThickness-1 );
+    double trgCellMipP = 0.;
+    if( thickCorr_.at( cellThickness-1 ) > 0 ){
+        trgCellMipP = amplitude / thickCorr_.at( cellThickness-1 ); 
+    }else{
+        edm::LogWarning("DivisionByZero") << "WARNING: the cell-thickness correction factor is zero";
```

Comment:
I would rather not print a warning there. Otherwise it will print it for each trigger cell.
Better to print this warning just once, in the constructor.